lib/src/domain/result/Result.ts

```diff
@@ -1,14 +1,22 @@
+import { Guards } from '../guards';
+
 export class Result<T> {
     public isSuccess: boolean;
-    public isFailure: boolean
+    public isFailure: boolean;
     public value: T | Error[];
 
+    public static guards = Guards;
```

Comment:
What is the idea of having Guards here inside Result?
The purpose of Result is to represent some execution output with error suppression
Guards it is a tooling to check some value boundaries.
IMHO
Guards should not be in Result. We can use it separately
Tha fact that you put it as static property witnesses that.

Reply:
All guards are returning Result instances, either it is success or failure, which is why we absolutely can't use those guards without Result. It seems like I misunderstood something

Reply:
Because result is the representation of check action and nothing more. So what is the connection between Result and Guard?
The main purpose of Result is representation of two possible endings of some abstract action. Can Result live without Guard? Definitely yes.
In other hand Guards is a set of tools to examine some value and it uses Result to represent result of this examination. So every Guard depends on Result.
Can we organise guard checking without Guards - yes. So Guards is a intention to separate repetitive blocks of similar checking code in one place and nothing more.
By making guards static property of Result we will create artificial useless circular dependency between those two objects.

Reply:
For example someone can do something like that (implement notISODateTime guard through itself)
export default (value: string): Result<string> | Result<Error> =>
    Result.guards.notISODateTime(value)
        ? Result.ok<string>(value)
        : Result.failure(new Error(`Invalid date value: ${value}`));

Reply:
Okay, now I see, thanks. I'll fix it